collection/rancher/v2.x/logs-collector/rancher2_logs_collector.sh

```diff
@@ -541,18 +545,33 @@ cleanup() {
 
 }
 
+upload() {
+
+  if $(command -v curl >/dev/null 2>&1); then
```

Comment:
I don't know how the FTP upload works. How do we handle collisions?

Reply:
Old files are overwritten without warning. Its good to have a unique prefix like the case number for filenames.

Reply:
@Thr3d Does the same applies to the folders? With the current logic if two customers enter the same case id (incorrectly, rare but still possible) then the newer upload will overwrite the old one. Is there some setting on the server-side where we can auto append some string or create a new folder for every upload so that we don't lose the files/uploads/data?

Reply:
If someone tried to upload a file of the same name to that directory it would overwrite the old file.

Not that I am aware of. It's rarely ever been an issue save for people that use very generic names when they upload files like "vmcore". Adding a random string to the name of files could be worthwhile for some but personally its not something I want to see. It would complicate customers' ability to communicate to us the names of files they upload.
Maybe if the server side PHP script checked for a case number in the filename, if not detected then suffix a random string to the end of the filename before the extension?

On the suse support side this hasn't been of much concern as most of the file names are unique enough, plus (should) have the case number.

Reply:
The issue with the PHP idea is the public side has HTTPS and FTP access so that would actually be difficult to accomplish.